### DISCUSSION TITLE
Terraform Plugin for Coolify V4

### DISCUSSION THREAD

**Discussion by maxexcloo** [coollabsio/coolify#2192] [Service Template Requests]
Hi all,

Wondering if there's any Coolify plugin for Terraform V4 (or plans for one) - have seen some for V3, would be awesome to add to my Terraform for my infra.

Thanks for an awesome project!

**Comment by boredland** [comment]
Would also be very interested in a Terraform provider, especially for secrets-management. The API looks quite fine already around that feature.

**Comment by SierraJC** [comment]
@boredland @maxexcloo
I'm actively working on a Terraform Provider for Coolify v4 @ https://registry.terraform.io/providers/SierraJC/coolify/latest/docs [ [git](https://github.com/SierraJC/terraform-provider-coolify) ]. It is not yet feature complete due to many Coolify OpenAPI schema quirks, but I'd love to know more about your desired use cases!

**Reply by boredland** [reply]
Our use-case would be mostly the management of secrets/environment variables of a coolify app.

**Reply by SierraJC** [reply]
@boredland 
> Our use-case would be mostly the management of secrets/environment variables of a coolify app.

Management of environment variables for [applications](https://registry.terraform.io/providers/SierraJC/coolify/latest/docs/resources/application_envs) / [services](https://registry.terraform.io/providers/SierraJC/coolify/latest/docs/resources/service_envs) has been added in [v0.5.0](https://github.com/SierraJC/terraform-provider-coolify/releases/tag/v0.5.0) 😃
Databases will be added next.

**Comment by DrakeJohnson6545** [comment]
@boredland @maxexcloo

There is now a more complete Terraform provider for Coolify v4: ⁠ coolify-terraform/coolify ⁠ (https://registry.terraform.io/providers/coolify-terraform/coolify/latest).

It covers 33 resources and 44 data sources (applications, databases, services, servers, environment variables, private keys, backups, scheduled tasks, GitHub Apps, S3 storage, and more). 880+ tests including acceptance tests against a real Coolify instance, OpenSSF Best Practices badge, and published on both the Terraform Registry and OpenTofu Registry.

If you are coming from the SierraJC provider, there is a migration guide: https://registry.terraform.io/providers/coolify-terraform/coolify/latest/docs/guides/migration-from-community

Quick start: https://registry.terraform.io/providers/coolify-terraform/coolify/latest/docs/guides/quickstart
